src/renderer/entities/network/model/network-model.ts

```diff
@@ -0,0 +1,250 @@
+    chainId,
+    nodes,
+    getMetadata,
+    () => {
+      console.info('🟢 provider connected ==> ', chainId);
+
+      networkConnected(chainId);
+    },
+    () => {
+      console.info('🔶 provider disconnected ==> ', chainId);
+
+      networkDisconnected(chainId);
+    },
+    () => {
+      console.info('🔴 provider error ==> ', chainId);
+
+      networkError(chainId);
+    },
+  );
+
+  return provider;
+});
+
+type CreateApiParams = {
+  chainId: ChainId;
+  provider: UniversalProvider;
+};
+const createApiFx = createEffect(async ({ chainId, provider }: CreateApiParams): Promise<ApiPromise | undefined> => {
+  try {
+    const api = await networkService.createApi(provider);
+
+    return api;
+  } catch (e) {
+    console.log('error during create api', e);
+  }
+});
+
+type ConnectParams = {
+  provider: UniversalProvider;
+  api: ApiPromise;
+  chainId: ChainId;
+};
+const connectFx = createEffect(async ({ provider, api, chainId }: ConnectParams): Promise<ChainId> => {
+  await networkService.connect(provider, api);
+
+  return chainId;
+});
+
+type DisconnectParams = {
+  provider: UniversalProvider;
+  api: ApiPromise;
+  chainId: ChainId;
+};
+const disconnectFx = createEffect(async ({ provider, api, chainId }: DisconnectParams): Promise<ChainId> => {
+  await networkService.disconnect(provider, api);
+
+  return chainId;
+});
+
+forward({
+  from: kernelModel.events.appStarted,
+  to: initConnectionsFx,
+});
+
+sample({
+  clock: initConnection,
+  fn: (chainId: ChainId) => ({
+    chainId,
+    nodes: chains[chainId].nodes.map((node) => node.url),
+  }),
+  target: createProviderFx,
+});
+
+sample({
+  source: $providers,
+  clock: createProviderFx.done,
+  fn: (providers, { params, result: provider }) => ({
+    ...providers,
+    [params.chainId]: provider,
+  }),
+});
+
+sample({
+  clock: createProviderFx.done,
+  fn: ({ params, result: provider }) => ({
+    chainId: params.chainId,
+    provider,
+  }),
+  target: createApiFx,
+});
+
+sample({
+  source: $apis,
+  clock: createApiFx.doneData,
+  filter: (api) => Boolean(api),
+  fn: (apis, api) => ({
+    ...apis,
+    [api!.genesisHash.toHex()]: api,
+  }),
+  target: $apis,
+});
+
+sample({
+  source: { providers: $providers, apis: $apis },
+  clock: networkConnectStarted,
+  fn: ({ providers, apis }, chainId) => ({
+    provider: providers[chainId],
+    api: apis[chainId],
+    chainId: chainId,
+  }),
+  target: connectFx,
+});
+
+sample({
+  source: { providers: $providers, apis: $apis },
+  clock: networkDisconnectStarted,
+  filter: ({ providers, apis }, chainId) => Boolean(providers[chainId] && apis[chainId]),
+  fn: ({ providers, apis }, chainId) => ({
+    provider: providers[chainId],
+    api: apis[chainId],
+    chainId: chainId,
+  }),
+  target: disconnectFx,
+});
+
+const switchProviderTypeFx = createEffect(
+  async ({ provider, type }: { provider: UniversalProvider; type: ProviderType }) => {
+    await provider.setProviderType(type);
```

Comment:
`public setProviderType(providerType: ProviderType) {` is a synchronous  function
So why do we call `await` here?